src/service/user_service.rs

```diff
@@ -58,13 +59,21 @@ pub async fn get_all_users(db_pool: web::Data<DbPool>) -> Result<Vec<User>, AppE
   .await;
 
   match result {
-    Ok(inner_result) => inner_result,
+    Ok(inner_result) => inner_result.map(|result_users| {
```

Comment:
Does it not map automatically with the From impl?

Reply:
Nope, rust wants all conversions to be declared explicitly